### DISCUSSION TITLE
Clarification on FlxSave

### DISCUSSION THREAD

**Discussion by Lbones-ai** [HaxeFlixel/flixel#2379] [Q&A] [answered]
I've done a bit of research about FlxSave, but I cannot find where the save data is stored when you bind a salveslot like this:
```
var _gameSave:FlxSave;
_gameSave = new FlxSave(); 		// initialize
_gameSave.bind("SaveFile1");		// bind to the named save slot

[...save some data...]

_gameSave.flush();
```
The information must be somewhere right? I even tried to specify a path but the file is nowhere to be seen..

Also I'm wondering.. for saving information in general, it's better to use FlxSave or to create a file directly with sys.FileSystem?

**Comment by RichardBray** [comment, marked as answer]
AFAIK for html5 the save data is saved in the browser, so if you use a different browser or incognito/private tab the data will be wiped, or different. I'm not sure if it's in the cookies or localStorage (or neither), never really looked into it.

If you want to actually see what data is saved, breakpoints and traces are your friend.

I think for most use cases FlxSave is fine. If you wanted to have an exported save file one that the user can move about at will I believe you need to serialize the data which is beyond my gamedev knowledge.

**Reply by Lbones-ai** [reply]
The main gripe I have with FlxSave is that I can't locate the save file on my PC, even if I gave the save slot a path the file is not there.

The other method I know is to use **lime.system.System** to get the documents path like so:
```haxe
_completePath = "" + System.documentsDirectory + "MySaveFolder";
if (FileSystem.exists(_completePath) == false) {
      FileSystem.createDirectory(_completePath);
}
```
That should work on any desktop application

For serializing the data I use this snippet, it seems to work just fine:
```haxe
typedef SaveDataMain = 
{
    @:optional var units:Array<SaveDataUnits>;
}

typedef SaveDataUnits =
{
    @:optional var posX:Int;
    @:optional var posY:Int;
}

[...]

var savUnits:Array<SaveDataUnits> = new Array<SaveDataUnits>();
for (i in 0 ... in_units.length) {
	savUnits.push( { posX: in_units[i].x, posY: in_units[i].y } );
}
var dat:SaveDataMain = { units: savUnits };
var serializer = new Serializer();
serializer.serialize(dat);
var s = serializer.toString();
trace(s);
```

**Reply by RichardBray** [reply]
Yeah seems like you have more experience in this than I do. I'll farm this question out to the discord server. There are more active people on that.

**Reply by RichardBray** [reply]
Okay we have some responses:
From @halfwheat: 
for hard disk on windows  it goes to `%USERPROFILE%\AppData\Roaming\HaxeFlixel\{ProjectName}`

From @NotBilly:
```
Also I'm wondering.. for saving information in general, it's better to use FlxSave or to create a file directly with sys.FileSystem?
```
For this question the answer is mostly... it doesn't matter at all. Whichever is possible and makes sense. But FlxSave is easier because it basically wraps both methodologies based on targe

**Reply by Lbones-ai** [reply]
> 
> 
> From @halfwheat:
> for hard disk on windows it goes to `%USERPROFILE%\AppData\Roaming\HaxeFlixel\{ProjectName}`

That's it, I haven't thought to check %appdata%, for me though is in `%USERPROFILE%\AppData\Roaming\%USERPROFILE%\{ProjectName}`

Will do more test with FlxSave, thanks man!
